tensorflow/compiler/mlir/tensorflow/transforms/lower_tf.td

```diff
@@ -252,6 +258,17 @@ def : Pat<(TF_RsqrtGradOp $lhs, $rhs),
                               (TF_ConstOp (GetScalarOfType<-2> $rhs))))>;
 
 
+//===----------------------------------------------------------------------===//
+// Size op patterns.
+//===----------------------------------------------------------------------===//
+
+// Size(x) = Prod(Shape(x), axis=0, keepdims=false)
+def : Pat<(TF_SizeOp:$result $arg),
+          (TF_ProdOp (CreateTFShapeOp $result, $arg, (IsI32 $result)),
```

Comment:
/*reduction_indices=*/(GetScalarOfType<0> $result)

As reduction_indices in prod to be of i32 if the elements are i32 and I think scalar reduction indices are allowed.